tests/unit/optimization/FirstOrderSAGCostFunctionInterface_unittest.cc

```diff
@@ -0,0 +1,146 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Elfarouk
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/optimization/FirstOrderSAGCostFunctionInterface.h>
+#include "FirstOrderSAGCostFunctionInterface_unittest.h"
+#include <Eigen/Dense>
+
+using namespace shogun;
+using Eigen::Matrix;
+using Eigen::Dynamic;
+using stan::math::var;
+using std::function;
+
+/** This is a temporary fix. The variables are now variables
```

Comment:
What does this mean?

Reply:
we need to figure out how we can store (if we want a none primitive typed std::vector, like stan::math:var)... hence the temporary story... just add a FIXME/TODO term there and then its good

Reply:
@iglesias  As Viktor said. The original cost function needs a *reference* to an SGVector<float64_t> of the parameters. The parameters have now been changed in the cost function to SGVector<stan::math::var> so we're going to have to change the interface of the minimizers before returning a SGVector<stan::math::var>,